code/datums/crew_credits.dm

```diff
@@ -0,0 +1,276 @@
+	var/list/antagonist = list()
+	var/list/captain = list()
+	var/list/security = list()
+	var/list/medical = list()
+	var/list/science = list()
+	var/list/engineering = list()
+	var/list/civilian = list()
+	var/list/silicon = list()
+	var/list/other = list()
+
+	for(var/datum/mind/M in ticker.minds)
+
+		// Antagonist?
+		if(M.special_role)
+			antagonist += bundle_crew_member_data(M)
+		if(!M.assigned_role)
+			continue
+
+		if (M.assigned_role == "Captain")
+			captain += bundle_crew_member_data(M)
+		else if ((M.assigned_role in security_jobs) || (M.assigned_role in security_gimmicks))
+			security += bundle_crew_member_data(M)
+		else if ((M.assigned_role in medical_jobs) || (M.assigned_role in medical_gimmicks))
+			medical += bundle_crew_member_data(M)
+		else if ((M.assigned_role in science_jobs) || (M.assigned_role in science_gimmicks))
+			science += bundle_crew_member_data(M)
+		else if ((M.assigned_role in engineering_jobs) || (M.assigned_role in engineering_gimmicks))
+			engineering += bundle_crew_member_data(M)
+		else if ((M.assigned_role in service_jobs) || (M.assigned_role in service_gimmicks) || M.assigned_role == "Staff Assistant")
+			civilian += bundle_crew_member_data(M)
+		else if ((M.assigned_role == "AI") || (M.assigned_role == "Cyborg"))
+			silicon += bundle_crew_member_data(M)
+		else if (M.assigned_role == "Pathologist")
+			#ifdef SCIENCE_PATHO_MAP
+			science += bundle_crew_member_data(M)
+			#else
+			medical += bundle_crew_member_data(M)
+			#endif
+		else
+			other += bundle_crew_member_data(M)
+
+	logTheThing(LOG_DEBUG, null, "Zamujasa/CREWCREDITS: [world.timeofday] done processing minds. info: A [antagonist.len] C [captain.len] S [security.len] M [medical.len] R [science.len] E [engineering.len] Cv [civilian.len] X [other.len]")
+
+	/* ~ BEGIN FAKE CREW CREDITS GENERATION ~ */
+	#ifdef CREDITS_DEBUGGING
+
+	var/has_head = FALSE
+	var/what_role = ""
+
+	while (antagonist.len < 4)
+		antagonist += src.generate_fake_crew_member(
+			real_name=src.fake_carbon_name(),
+			role = pick("Vampire", "Werewolf", "Changeling", "Legworm", "Handspider", "Eyespider", "Traitor", "Spy-thief", "Blob", "Flockmind", "Flockbit", "Omnitraitor", "Nuclear Operative", "Hard-mode Traitor"),
+			dead = prob(50),
+		)
+	while (captain.len < 1)
+		captain += src.generate_fake_crew_member(
+			real_name=src.fake_carbon_name(),
+			role= "Captain",
+			dead=prob(20),
+			head=TRUE
+		)
+	while (security.len < 8)
+		if (!has_head)
+			what_role = "Head of Security"
+		else
+			what_role = pick("Security Officer", "Security Assistant")
+		security += src.generate_fake_crew_member(
+			real_name=src.fake_carbon_name(),
+			role=what_role,
+			dead=prob(30),
+			head=!has_head
+		)
+		has_head=TRUE
+	has_head=FALSE
+	while (medical.len < 8)
+		if (!has_head)
+			what_role = "Medical Director"
+		else
+			what_role = pick("Medical Doctor", "Medical Doctor", "Roboticist", "Geneticist") // weighted (for "realism")
```

Comment:
doesn't really matter but you can weight `pick` without duplicating entries https://secure.byond.com/docs/ref/#/proc/pick

Reply:
[47367ac](https://github.com/goonstation/goonstation/pull/12339/commits/47367acf3d80acceb8a5a6424107802322029292)